monai/losses/aucm_loss.py

```
@@ -0,0 +1,172 @@
+
+        self.margin = margin
+
+        self.p = pos_weight
+        self.num_classes = num_classes
+        if self.p:
+            assert len(self.p) == self.num_classes, "imratio must be a list of length num_classes"
+        else:
+            self.p = [0] * self.num_classes
+
+        self.to_onehot_y = to_onehot_y
+        self.sigmoid = sigmoid
+        self.softmax = softmax
+        self.other_act = other_act
+
+        self.a = torch.zeros(self.num_classes, dtype=torch.float32, requires_grad=True)
+        self.b = torch.zeros(self.num_classes, dtype=torch.float32, requires_grad=True)
+        self.alpha = torch.zeros(self.num_classes, dtype=torch.float32, requires_grad=True)
+
+    def forward(self, y_pred: torch.Tensor, y_true: torch.Tensor, auto=True) -> torch.Tensor:
+        """
+        Args:
+            y_pred: predicted tensor of shape [batch_size, num_classes, ...]
+            y_true: ground truth tensor of shape [batch_size, num_classes, ...]
+            auto: if True, automatically calculate imbalance ratio. Defaults to True.
+
+        Raises:
+            ValueError: When y_pred and num_classes have different sizes.
+            ValueError: When y_true and num_classes have different sizes.
+            ValueError: When y_pred and y_true have different sizes.
+        """
+        if y_pred.shape[1] != self.num_classes:
+            raise ValueError(
+                f"y_pred.shape[1] must be {self.num_classes}, equal to num_classes ,but is {y_pred.shape[1]}"
+            )
+
+        if y_true.shape[1] != self.num_classes:
+            raise ValueError(
+                f"y_true.shape[1] must be {self.num_classes}, equal to num_classes ,but is {y_true.shape[1]}"
+            )
+
+        if self.sigmoid:
+            y_pred = torch.sigmoid(y_pred)
+
+        n_pred_ch = y_pred.shape[1]
+        if self.softmax:
+            if n_pred_ch == 1:
+                warnings.warn("single channel prediction, `softmax=True` ignored.")
```

Isn't it better to raise an error here instead of a warning?
Considering that `softmax` is defaulted to `False`, it means that the user has explicitly set it to `True` and the the input argument is wrongly set, so it might be better to raise a ValueError. Does it make sense?